packages/verified-fetch/test/cache-control-header.spec.ts

```diff
@@ -0,0 +1,131 @@
+  const fake = stubInterface<DNSResponse>()
+  fake.Answer = [{
+    data,
+    TTL,
+    name,
+    type
+  }]
+  return fake
+}
+describe('cache-control header', () => {
+  let helia: Helia
+  let name: IPNS
+  let verifiedFetch: VerifiedFetch
+  let customDnsResolver: Sinon.SinonStub<any[], Promise<DNSResponse>>
+
+  beforeEach(async () => {
+    customDnsResolver = Sinon.stub()
+    helia = await createHelia({
+      dns: dns({
+        resolvers: {
+          '.': customDnsResolver
+        }
+      })
+    })
+    name = ipns(helia)
+    verifiedFetch = new VerifiedFetch({
+      helia
+    })
+  })
+
+  afterEach(async () => {
+    await stop(helia, verifiedFetch)
+  })
+
+  it('should allow return the correct max-age in the cache header for immutable responses', async () => {
+    const obj = {
+      hello: 'world'
+    }
+    const c = dagCbor(helia)
+    const cid = await c.add(obj)
+
+    const resp = await verifiedFetch.fetch(cid)
+
+    expect(resp).to.be.ok()
+    expect(resp.status).to.equal(200)
+    expect(resp.headers.get('Cache-Control')).to.equal('public, max-age=29030400, immutable')
+  })
+
+  it('should return not contain immutable in the cache-control header for an IPNS name', async () => {
+    const obj = {
+      hello: 'world'
+    }
+    const c = dagCbor(helia)
+    const cid = await c.add(obj)
+
+    const oneHourInMs = 1000 * 60 * 60
+    const peerId = await createEd25519PeerId()
+
+    // ipns currently only allows customising the lifetime which is also used as the TTL
+    await name.publish(peerId, cid, { lifetime: oneHourInMs })
+
+    const resp = await verifiedFetch.fetch(`ipns://${peerId}`)
+    expect(resp).to.be.ok()
+    expect(resp.status).to.equal(200)
+
+    expect(resp.headers.get('Cache-Control')).to.not.containIgnoreCase('immutable')
+  })
+
+  it('should return the correct max-age in the cache-control header for an IPNS name', async () => {
+    const obj = {
+      hello: 'world'
+    }
+    const c = dagCbor(helia)
+    const cid = await c.add(obj)
+
+    const oneHourInSeconds = 60 * 60
+    const peerId = await createEd25519PeerId()
+
+    /**
+     * ipns currently only allows customising the lifetime which is also used as the TTL
+     *
+     * lifetime is coming back as 100000 times larger than expected
+     *
+     * @see https://github.com/ipfs/js-ipns/blob/16e0e10682fa9a663e0bb493a44d3e99a5200944/src/index.ts#L200
+     * @see https://github.com/ipfs/js-ipns/pull/308
+     */
+    await name.publish(peerId, cid, { lifetime: oneHourInSeconds * 1000 }) // pass to ipns as milliseconds
+
+    const resp = await verifiedFetch.fetch(`ipns://${peerId}`)
+    expect(resp).to.be.ok()
+    expect(resp.status).to.equal(200)
+
+    expect(resp.headers.get('Cache-Control')).to.equal(`public, max-age=${oneHourInSeconds}`)
+  })
+
+  it('should not contain immutable in the cache-control header for a DNSLink name', async () => {
+    verifiedFetch = new VerifiedFetch({
```

Comment:
The code above (https://github.com/ipfs/helia-verified-fetch/pull/19/files#diff-853a9f8afb602ec83fff59dd64982c5e2adf008e30cdcf9c179fe361f0455c48R32-R43) already instantiates verifiedFetch with the customDnsResolver, so it should be needed here.

Reply:
verifiedFetch in `beforeEach` doesn't pass dnsResolvers, so this fails without it.

Reply:
It has the Helia instance passed to it with the customDnsResolver 

https://github.com/ipfs/helia-verified-fetch/pull/19/files#diff-853a9f8afb602ec83fff59dd64982c5e2adf008e30cdcf9c179fe361f0455c48R36

Reply:
you're right.. removing